cfme/scripting/appliance.py

```diff
@@ -29,6 +33,57 @@ def main():
     pass
 
 
+@main.command('upgrade', help='Upgrades an appliance to new version')
+@click.argument('appliance-ip', default=None, required=False)
+@click.option('--cfme-only', is_flag=True, help='Upgrade cfme packages only')
+@click.option('--update-to', default='59', help='Must be 58 or (59 is default)')
+def upgrade_appliance(appliance_ip, cfme_only, update_to):
+    """Upgrades an appliance"""
+    if appliance_ip:
+        print 'Connecting to {}'.format(appliance_ip)
+    else:
+        print 'Fetching appliance from env.local.yaml'
+    app = get_appliance(appliance_ip)
+    update_url = 'update_url_{}'.format(update_to)
+    print 'Extending appliance partitions'
+    app.db.extend_partition()
+    urls = process_url(cfme_data['basic_info'][update_url])
+    output = build_file(urls)
+    print 'Adding update repo to appliance'
+    with tempfile.NamedTemporaryFile('w') as f:
+        f.write(output)
+        f.flush()
+        os.fsync(f.fileno())
+        app.ssh_client.put_file(
+            f.name, '/etc/yum.repos.d/update.repo')
+    ver = '95' if app.version >= '5.8' else '94'
+    cfme = '-y'
+    if cfme_only:
+        cfme = 'cfme -y'
+    print 'Stopping EVM'
+    app.evmserverd.stop()
+    with app.ssh_client as ssh:
+        print 'Running yum update'
+        rc, out = ssh.run_command('yum update {}'.format(cfme), timeout=3600)
+        assert rc == 0, "update failed {}".format(out)
```

Comment:
**REQUIRED**
You can extend `run_command` function from [here](https://github.com/ManageIQ/integration_tests/blob/master/cfme/scripting/setup_ansible.py#L43) and use one line instead of three ones.